### PR TITLE
Support runtime as an input

### PR DESCRIPTION
**Pull request by RodrigoEspinosa**
Issue: https://github.com/serverless-components/backend/issues/1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ Easily host entire web applications on a single AWS Lambda function using this [
 
 ### Features
 
-* Designed to make it easy to host pre-existing web frameworks (e.g. Express.js, Hapi) or any large web application on a single AWS Lambda Function.
-* Blazing Fast Uploads via AWS S3 Accelerated Transfer and Multi-Part.
-* Dependencies are automatically put in AWS Lambda Layers, reducing cold-start time and further reducing upload time.
-* Simple shim for receiving and responding to HTTP requests.
-* Supports specifying custom domains.
+- Designed to make it easy to host pre-existing web frameworks (e.g. Express.js, Hapi) or any large web application on a single AWS Lambda Function.
```
Comment (RodrigoEspinosa) [inline README.md:9]: I'm sorry for this, I notice that they were automatically changed by the vscode linter. That said, I notice another code inconsistencies in `serverless.types.js`, please let me know if you want me to create another PR for that.

File: serverless.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ class Backend extends Component {
       description: inputs.description || 'A function for the Backend Component',
       memory: inputs.memory || 896,
       timeout: inputs.timeout || 10,
-      runtime: 'nodejs8.10',
+      runtime: inputs.runtime || 'nodejs10.x',
```
Comment (eahefnawy) [inline serverless.js:102]: @RodrigoEspinosa we should really throw a user-friendly error if it's not a node runtime, cause only node is supported due to the shim

**Review by eahefnawy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (eahefnawy) [reply README.md:9]: No worries 😊

File: serverless.js (hunk 2 above)
Comment (RodrigoEspinosa) [reply serverless.js:102]: Good idea, I'll make sure to include it today, thanks!

Comment (RodrigoEspinosa) [reply serverless.js:102]: Would something like this make sense, or making the supported runtimes dynamic isn't necessary at this point?

```js
// Get a list of accepted runtimes
const acceptedRuntimes =
  this.types().functions.default.inputs.find((input) => input.name === 'code').runtimes || []

// Check if the runtime from the input is not accepted
if (inputs.runtime && !acceptedRuntimes.findIndex(inputs.runtime)) {
  throw new Error(
    `The runtime ${inputs.runtime} is not accepted. Currently only ${acceptedRuntimes.join(
      ', '
    )} are supported.`
  )
}
```

Comment (eahefnawy) [reply serverless.js:102]: Imo it's not necessary at this point. The types that you're referencing to is still experimental and we don't currently rely on it.

Comment (RodrigoEspinosa) [reply serverless.js:102]: Sorry for the delay, it's there now. Let me know if you want more changes there.

Comment (eahefnawy) [reply serverless.js:102]: Thanks @RodrigoEspinosa ... but I don't see the error handling logic! 🤔 ... what would happen if I specify a runtime other than node?

**Review by eahefnawy — CHANGES_REQUESTED** [review]

File: serverless.js (hunk 2 above)
Comment (RodrigoEspinosa) [reply serverless.js:102]: Whoops! My apologies. Should be ok now, let me know if you the message to say something else.

Here it is: https://github.com/serverless-components/backend/pull/4/files#diff-442235eb56754d8a737ee4aaf1cefd62R67-R71

**Review by eahefnawy — APPROVED** [review]
Thanks @RodrigoEspinosa 🙌

**Pull request merged**